game/squadrons/airwing.py

```diff
@@ -57,9 +58,15 @@ def best_squadrons_for(
     def best_squadron_for(
         self, location: MissionTarget, task: FlightType, size: int, this_turn: bool
     ) -> Optional[Squadron]:
-        for squadron in self.best_squadrons_for(location, task, size, this_turn):
-            return squadron
-        return None
+        best_squadrons = self.best_squadrons_for(location, task, size, this_turn)
+        if not best_squadrons:
+            return None
+        else:
+            # In case we have several best squadrons at this point,
+            # return a random squadron from the list of best squadrons for this MissionTarget
+            # so the autoplanner will reserve pilots and aircraft from several squadrons,
+            # not just the first one in the list.
+            return random.choice(best_squadrons)
```

Comment:
What does this have to do with the rest of the PR?

Reply:
Currently, the autoplanner always seems to select the same squadron every time if the best_squadrons_for list has the same contents. This leads it to make purchases which are not evenly distributed across e.g. other squadrons operating out of the same base, with the same aircraft type and with identical task profiles:
![image](https://user-images.githubusercontent.com/89945461/145710105-8118df03-d5cd-4823-86bc-78142166c53d.png)

In the above screenshot, there have been no combat losses. The differences in the aircraft counts are purely due to the best squadron selection and the procurement AI buying aircraft based on it.

My concern was that if I limit the procurement of aircraft to the max pilot limit, substituting purchases would simply be scrubbed instead of buying the aircraft for other capable squadrons, without this added random selection.

Reply:
That's the intent. The auto planner is supposed to make good decisions, not random ones.